### ISSUE TITLE
Add a Helm chart for the operator deployment

### ISSUE THREAD

**Issue by liyinan926** [GoogleCloudPlatform/spark-on-k8s-operator#79]

**Comment by skonto** [comment]
Should be added here: https://github.com/kubernetes/charts?

**Comment by liyinan926** [comment]
Yeah, it could potentially be the place. I guess it will start as an incubator chart?

**Comment by skonto** [comment]
Ok will have a look, thanx.

**Comment by yuchaoran2011** [comment]
A [PR](https://github.com/kubernetes/charts/pull/6239) is now available at Kubernetes charts repo.

**Comment by liyinan926** [comment]
Thanks, @yuchaoran2011!

**Comment by mrow4a** [comment]
It would be interesting to use sth like this for webhooks https://github.com/kubernetes/charts/blob/37bf2e2867a124219ce844fd9645fd8f73816583/stable/dex/templates/job.yaml in future Helm Chart. Thoughts?

**Comment by liyinan926** [comment]
@mrow4a Just saw your comment on using a Job to generate the certificate. I like the idea.  @yuchaoran2011 it would be awesome if you can add it into your PR. Thanks!

**Comment by yuchaoran2011** [comment]
@liyinan926 Ah that's a good idea! I'm on vacation now but I'll update the PR when I'm back.

**Comment by mrow4a** [comment]
@liyinan926 I can also try to figure that out - lets maybe merge what is there first, the other PR is already quite fat.

**Comment by mrow4a** [comment]
@yuchaoran2011 perfect!

**Comment by mrow4a** [comment]
@liyinan926 In fact we can add it as another YAML in manifest right?

**Comment by liyinan926** [comment]
@mrow4a @yuchaoran2011 

> I can also try to figure that out - lets maybe merge what is there first, the other PR is already quite fat.

Sounds good.

> In fact we can add it as another YAML in manifest right?

Yes, I think so.

**Comment by yuchaoran2011** [comment]
@mrow4a @liyinan926 Now I have a Docker image that bundles a customized version of gencerts.sh. Then I define a k8s Job in YAML that runs the custom gencerts.sh to generate and install the secret. But the problem is that Helm does not enforce a particular order in which the YAML files are applied, so k8s resources that require the webhook secret can be applied before the secret is created, causing a failure in the chart installation process. I tried making the Job a pre-install hook. But that didn't work because the Job relies on the sparkoperator serviceAccount defined in another YAML file (it uses the token corresponding to the serviceAccount to talk to the API server to install the secret). Any suggestions how I should approach this issue?

**Comment by mrow4a** [comment]
@yuchaoran2011 I think it has to somehow, since you cannot e.g. create spark operator without e.g. sparkoperator namespace. Could you share your progress (changed files) so we could review (WIP)?

**Comment by yuchaoran2011** [comment]
@mrow4a Sure. I'll clean up things and make a PR today for you guys to review.

**Comment by yuchaoran2011** [comment]
@mrow4a @liyinan926 The [PR](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/247) is now ready. I also made changes to the Helm chart [PR](https://github.com/helm/charts/pull/6239#issuecomment-411062539). Please review both places when you get time. Thanks!

**Comment by yuchaoran2011** [comment]
@liyinan926 Now that the PR is merged, would you want me to update the operator doc to point to the chart as the way to install? Do we still want to keep the `/manifest` directory?
